### PR TITLE
SQL generator for stable views

### PR DESCRIPTION
**Pull request by scholtzan**
Depends on https://github.com/mozilla/bigquery-etl/pull/2567
For #2313

This moves the stable view generation into `sql_generators/` so stable views can be generated via `./bqetl generate stable_views`

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: bigquery_etl/schema/stable_table_schema.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
+"""Methods for working with stable table schemas."""
```
Comment (scholtzan) [inline bigquery_etl/schema/stable_table_schema.py:1]: Moved out from `bigquery_etl/view/generate_stable_views.py`. These methods are used in a couple of places.

File: sql/moz-fx-data-shared-prod/org_mozilla_fenix/baseline_clients_daily/metadata.yaml [hunk 2]
```diff
@@ -1,12 +0,0 @@
----
-# Generated via bigquery_etl.glean_usage
```
Comment (scholtzan) [inline sql/moz-fx-data-shared-prod/org_mozilla_fenix/baseline_clients_daily/metadata.yaml:2]: These files should get created as part of the SQL generation afaik, so no need to check them in to `main`

File: sql_generators/feature_usage/__init__.py [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 
+import click
+import yaml
 from jinja2 import Environment, FileSystemLoader
-from pathlib import Path
 
 from bigquery_etl.cli.query import update
-from bigquery_etl.util.common import write_sql
 from bigquery_etl.format_sql.formatter import reformat
+from bigquery_etl.util.common import write_sql
```
Comment (scholtzan) [inline sql_generators/feature_usage/__init__.py:12]: I ran `black` on `sql_generators/`, so there are a couple of formatting changes in those files

**Review by jklukas — APPROVED** [review]
Looks good, including the refactorings.

This is a case where it would be great to see the diff in generated content. If we rebase, will this PR get a diff?

**Comment by jklukas** [comment]
Looks like the diff posting silently failed. See https://github.com/mozilla/bigquery-etl/issues/2620

**Comment by scholtzan** [comment]
Depends on https://github.com/mozilla/bigquery-etl/pull/2624

File: sql_generators/stable_views/__init__.py [hunk 4]
```diff
@@ -9,36 +9,20 @@
...
-
-SKIP_PREFIXES = (
-    "pioneer",
-    "rally",
-)
+from bigquery_etl.schema.stable_table_schema import SchemaFile, get_stable_table_schemas
 
 VIEW_QUERY_TEMPLATE = """\
--- Generated by bigquery_etl.view.generate_stable_views
+-- Generated via ./bqetl generate stable_views
```
Comment (scholtzan) [inline sql_generators/stable_views/__init__.py:25]: Okay, the diff we are seeing makes sense. I changed the comment here.

**Review by jklukas — APPROVED** [review]

**Pull request merged**